bot/exts/recruitment/talentpool/_cog.py

```diff
@@ -122,70 +128,128 @@ async def autoreview_loop(self) -> None:
             log.info("Running check for users to nominate.")
             await self.reviewer.maybe_review_user()
 
-    @nomination_group.command(
-        name="nominees",
-        aliases=("nominated", "all", "list", "watched"),
-        root_aliases=("nominees",)
+    @tasks.loop(hours=24)
+    async def prune_talentpool(self) -> None:
+        """
+        Prune any inactive users from the talentpool.
+
+        A user is considered inactive if they have sent no messages on the server
+        in the past `DAYS_UNTIL_INACTIVE` days.
+        """
+        log.info("Running task to prune users from talent pool")
+        nominations = await self.api.get_nominations(active=True)
+
+        if not nominations:
+            return
+
+        messages_per_user = await self.api.get_activity(
+            [nomination.user_id for nomination in nominations],
+            days=DAYS_UNTIL_INACTIVE
+        )
+
+        nomination_discussion = self.bot.get_channel(Channels.nomination_discussion)
```

Comment:
Wouldn't this fail if the channel is not in cache ?

If that's the case, then we might end up with a null value for that channel .

We could either add our own `get_or_fetch` to bot-core, or have a null-check here.


Danny said that `get_or_fetch` is not there on purpose, here's the [issue](https://github.com/Rapptz/discord.py/issues/9213) for reference.

Reply:
Normal text channels (i.e. not threads) *should* always be cached, it's something we assume quite a lot throughout the bot.

I do agree that this isn't ideal, although i'm not sure what the nicest approach is. At the very least it's annoying to have my editor complain that it could be `None`.

Since it's something that shouldn't be `None`, I don't think we should just ignore if it is (e.g. early return), as this could hide some bugs.

One solution is something i'd thought of when thinking about how to improve type-hintedness (https://github.com/python-discord/bot/issues/1773#issuecomment-1251021265). This would involve creating our own method on `bot` which raises an error if the channel can't be found instead of returning `None`. We could even extend this and provide e.g. a `get_textchannel` method which ensures that the channel is a textchannel and raises an error if it isn't, giving us a more precise type in the rest of the code (otherwise editors will still complain since e.g. voicechannels don't have send).

Reply:
Ah I didn't realise our get_or_fetch channel already never returned `None`, i'll use that https://bot-core.pythondiscord.com/main/output/pydis_core.utils.channel.html#pydis_core.utils.channel.get_or_fetch_channel